zarr/core.py

```diff
@@ -174,6 +174,8 @@ def __init__(
         self._attrs = Attributes(
             store, key=akey, read_only=read_only, synchronizer=synchronizer, cache=cache_attrs
         )
+        if cache_attrs:
+            self._attrs._cached_asdict = self._meta["attributes"]
```

Comment:
It seems like this optimization really belongs in the `Attributes.__init__` method. What do you think about passing the attributes through like this?

```suggestion
        self._attrs = Attributes(
            store, key=akey, read_only=read_only, synchronizer=synchronizer, cache=cache_attrs,
            cache_init=self._meta["attributes"]
        )
        if cache_attrs:
            self._attrs._cached_asdict = self._meta["attributes"]
```

Reply:
Would be fine, wasn't sure if you wanted to preserve the signature for some reason. Also there is some error in the tests, presumably v2 vs v3?

Reply:
It's there for arrays on V3 I think.